irr/src/CMakeLists.txt

```diff
@@ -460,19 +435,38 @@ add_library(IRRGUIOBJ OBJECT
 
 # Library
 
-add_library(IrrlichtMt STATIC)
-foreach(object_lib
-	IRRMESHOBJ IRROBJ IRRVIDEOOBJ
-	IRRIOOBJ IRROTHEROBJ IRRGUIOBJ)
-	# Set include directories for object library compilation
-	target_include_directories(${object_lib} PRIVATE ${link_includes})
-	# Add objects from object library to main library
-	target_sources(IrrlichtMt PRIVATE $<TARGET_OBJECTS:${object_lib}>)
-
-	if(BUILD_WITH_TRACY)
-		target_link_libraries(${object_lib} PRIVATE Tracy::TracyClient)
-	endif()
-endforeach()
+add_library(IrrlichtMt STATIC
+	${IRRLICHTMT_SOURCES}
```

Comment:
why are we getting rid of object libraries here?
this is a step back.

Reply:
Because of the CMake Xcode project generator.

Reply:
Sorry but that's a poor justification.

IMO if Xcode and/or CMake do not support this feature then we need to find a different way or wait for it to be supported, not stop using object libraries.

Reply:
CMake will generate non working project if object libraries are used in the Irrlicht case. It just did not create `libirrlichtmt.a` and the final link failed. It looks like a CMake Xcode generator to understand object libraries as static libraries.
That [problem](https://gitlab.kitware.com/cmake/cmake/-/issues/17500) looks to have been known at least for 6 years and still not fixed. So, I do not expect it to be fixed anytime soon.

Applied the fix we discussed on IRC.